tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/passes/BoundedCheckerPassImpl.scala

```diff
@@ -83,7 +81,11 @@ class BoundedCheckerPassImpl @Inject() (val options: PassOptions, hintsStore: Fo
     val tuning = options.getOrElse[Map[String, String]]("general", "tuning", Map[String, String]())
     val debug = options.getOrElse[Boolean]("general", "debug", false)
     val saveDir = options.getOrError[Path]("io", "outdir").toFile
-    val smtEncoding = options.get[Option[String]]("checker", "smt-encoding").flatten.getOrElse(oopsla19Encoding)
+    val smtEncoding = options.get[Option[String]]("checker", "smt-encoding").flatten.getOrElse("oopsla19") match {
+      case "oopsla19"      => oopsla19Encoding
+      case "arrays"        => arraysEncoding
+      case oddEncodingType => throw new IllegalArgumentException(s"Unexpected SMT encoding type $oddEncodingType")
+    }
```

Comment:
My expectation would be that instead of saving this as a string in `options`, we would save it the enum type, so that we could here just do `get[Option[SMTEncodings]]`. So, ideally, I think we should push the conversion from string into the cli parser. Documentation on extending the CLI parser to read these arguments into the target type are here: https://github.com/backuity/clist#parsing

I'd be be happy to sync up to discuss how to apply this approach!

Reply:
Thanks for the suggestion! You did mention the "Parse, Don't Validate" post to me before, but I guess haven't internalized it yet. I moved the check from `BoundedCheckerPassImpl` to `Tool`.

I wanted to have it in `CheckCmd`, but it seems that the CLI parses doesn't support this. I found the following comment on the [clist repo](https://github.com/backuity/clist/blob/master/core/src/main/scala/org/backuity/clist/util/Read.scala): "Note that we do not support scala Enumeration as this would require having a runtime dependency on the reflection API. On the other hand Java enums are supported.".

Reply:
Ok. That's not too bad. A solution might be to make this a case class, but this is already pretty far out in the edge of the program :)